hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/HoodieIncrSource.java

```diff
@@ -172,10 +178,13 @@ public Pair<Option<Dataset<Row>>, String> fetchNextBatch(Option<String> lastCkpt
      *
      * log.info("Validated Source Schema :" + validated.schema());
      */
+    boolean dropAllMetaFields = props.getBoolean(Config.HOODIE_DROP_ALL_META_FIELDS_FROM_SOURCE,
+        Config.DEFAULT_HOODIE_DROP_ALL_META_FIELDS_FROM_SOURCE);
 
     // Remove Hoodie meta columns except partition path from input source
-    final Dataset<Row> src = source.drop(HoodieRecord.HOODIE_META_COLUMNS.stream()
-        .filter(x -> !x.equals(HoodieRecord.PARTITION_PATH_METADATA_FIELD)).toArray(String[]::new));
+    String[] colsToDrop = dropAllMetaFields ? HoodieRecord.HOODIE_META_COLUMNS.stream().toArray(String[]::new) :
+        HoodieRecord.HOODIE_META_COLUMNS.stream().filter(x -> !x.equals(HoodieRecord.PARTITION_PATH_METADATA_FIELD)).toArray(String[]::new);
```

Comment:
Do we know what was the reason we strictly dropped partition path in the first place?

Reply:
I could not decode that. I don't see a need unless we want to carry over the partitioning from tableA to tableB.